### PR TITLE
Move OCS route /config to proper controller

### PR DESCRIPTION
**Pull request by rullzer**
I just copied the current code. Altough it might make sense to change some stuff.... but to what?

CC: @LukasReschke @MorrisJobke @nickvergessen @icewind1991

### CODE REVIEW

File: core/Controller/OCSController.php [hunk 1]
```diff
@@ -53,6 +53,23 @@ public function __construct($appName,
...
 
 	/**
+	 * @PublicPage
+	 *
+	 * @return DataResponse
+	 */
+	public function getConfig() {
+		$data = [
+			'version' => '1.7',
+			'website' => 'ownCloud',
```
Comment (icewind1991) [inline core/Controller/OCSController.php]: ?

Comment (nickvergessen) [inline core/Controller/OCSController.php]: > I just copied the current code.

**Comment by nickvergessen** [comment]
:+1:

File: core/Controller/OCSController.php (hunk 1 above)
Comment (rullzer) [inline core/Controller/OCSController.php]: yeah we could put Nextcloud there... as far as I know this route is not actively used...

Comment (rullzer) [inline core/Controller/OCSController.php]: done

**Comment by MorrisJobke** [comment]
It says: "CSRF check failed". Is this even needed, @LukasReschke?

File: core/Controller/OCSController.php [hunk 2]
```diff
@@ -53,6 +53,23 @@ public function __construct($appName,
 	}
 
 	/**
+	 * @PublicPage
+	 *
+	 * @return DataResponse
+	 */
+	public function getConfig() {
+		$data = [
+			'version' => '1.7',
```
Comment (MorrisJobke) [inline core/Controller/OCSController.php]: What version is this?

File: core/Controller/OCSController.php [hunk 3]
```diff
@@ -53,6 +53,23 @@ public function __construct($appName,
...
+			'website' => 'Nextcloud',
+			'host' => $this->request->getServerHost(),
```
Comment (MorrisJobke) [inline core/Controller/OCSController.php]: You are already contacting this host ... no need to get it again.

File: core/Controller/OCSController.php [hunk 4]
```diff
@@ -53,6 +53,23 @@ public function __construct($appName,
...
+			'contact' => '',
```
Comment (MorrisJobke) [inline core/Controller/OCSController.php]: Empty contact  makes much sense.

**Comment by LukasReschke** [comment]
> It says: "CSRF check failed". Is this even needed, @LukasReschke?

In short, the current API design does require it to be really secure. You have to send the OCS headers to make that pass properly. 🙈 – Session auth is biting us here a lot.

**Comment by MorrisJobke** [comment]
I think this is mostly dead code. I think it's because it is included in the spec right?

This reminds me of the blog about "killing broken features because they aren't event used at all": http://www.tedunangst.com/flak/post/broken-features-arent-used

**Comment by rullzer** [comment]
All ocs spec. We can kill it as well. I don't care. I just want to get it our of that static route file

File: core/Controller/OCSController.php [hunk 5]
```diff
@@ -53,6 +53,23 @@ public function __construct($appName,
...
+			'ssl' => 'false',
```
Comment (rullzer) [inline core/Controller/OCSController.php]: just copied over

File: core/Controller/OCSController.php (hunk 2 above)
Comment (rullzer) [inline core/Controller/OCSController.php]: OCS version

File: core/Controller/OCSController.php (hunk 3 above)
Comment (rullzer) [inline core/Controller/OCSController.php]: Again just copied the code

File: core/Controller/OCSController.php (hunk 4 above)
Comment (rullzer) [inline core/Controller/OCSController.php]: also just copied

**Comment by MorrisJobke** [comment]
:+1: for moving it. We can discuss about the content later.

**Pull request merged**